Sources/Hub/Downloader.swift

```diff
@@ -83,31 +91,28 @@ class Downloader: NSObject, ObservableObject {
             }
         }
         semaphore.wait()
-        
+
         switch downloadState.value {
         case .completed(let url): return url
         case .failed(let error):  throw error
         default:                  throw DownloadError.unexpectedError
         }
     }
-    
+
     func cancel() {
         urlSession?.invalidateAndCancel()
     }
 }
 
-extension Downloader: URLSessionDelegate, URLSessionDownloadDelegate {
+extension Downloader: URLSessionDownloadDelegate {
     func urlSession(_: URLSession, downloadTask: URLSessionDownloadTask, didWriteData _: Int64, totalBytesWritten _: Int64, totalBytesExpectedToWrite _: Int64) {
         downloadState.value = .downloading(downloadTask.progress.fractionCompleted)
     }
 
     func urlSession(_: URLSession, downloadTask _: URLSessionDownloadTask, didFinishDownloadingTo location: URL) {
-        guard FileManager.default.fileExists(atPath: location.path) else {
-            downloadState.value = .failed(DownloadError.invalidDownloadLocation)
-            return
-        }
         do {
-            try FileManager.default.moveItem(at: location, to: destination)
+            // If the downloaded file already exists on the filesystem, overwrite it
+            try FileManager.default.moveDownloadedFile(from: location, to: self.destination)
```

Comment:
I noticed if some files failed to download, it would never attempt to overwrite the existing files, which shouldn't really matter, but this enforces that the blob being downloaded "currently" has all the same files from that version. Future work to be done here related to cache invalidation #39